osi_accounting_enhancements/models/account_invoice.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    supplier_invoice_number = fields.Char(
+        string="Supplier Invoice Number",
+        help="The reference of this invoice as provided by the supplier.",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+
+    _sql_constraints = [
+        (
+            "supplier_invoice_number_uniq",
+            "unique(supplier_invoice_number,partner_id)",
+            "Supplier Invoice Number must be unique per Supplier!",
+        )
+    ]
+
+    @api.multi
+    def copy(self, default=None):
+        default = dict(
+            default or {},
+            name=_("%s (copy)") % self.name,
+            supplier_invoice_number=False,
+        )
+        return super(AccountInvoice, self).copy(default=default)
+
+    @api.model
+    def _prepare_refund(
+        self, invoice, date_invoice=None, date=None, description=None, journal_id=None
+    ):
+        values = super(AccountInvoice, self)._prepare_refund(
+            invoice,
+            date_invoice=date_invoice,
+            date=date,
+            description=description,
+            journal_id=journal_id,
+        )
+        for field in ["supplier_invoice_number", "date_due"]:
+            if field == "supplier_invoice_number":
+                if invoice[field]:
+                    values[field] = "Refund-{}-{}".format(description, invoice[field])
+                else:
+                    values[field] = "Refund-{}-{}".format(description, self.number)
+            else:
+                values[field] = invoice[field] or False
+        return values
+
+    # Commented By MG:05/21/2020 -- Don't want Supplier Invoice unique/mandatory
```

Comment:
@max3903 : I commented it because ART didn't want it, I think we should make it as a standardized module in osi-addons.